src/HotChocolate/Fusion-vnext/test/Fusion.Composition.Tests/PreMergeValidation/Rules/ExternalArgumentsDefaultMismatchRuleTests.cs

```diff
@@ -0,0 +1,140 @@
+using HotChocolate.Fusion;
+using HotChocolate.Fusion.Logging;
+using HotChocolate.Fusion.PreMergeValidation;
+using HotChocolate.Fusion.PreMergeValidation.Rules;
+using HotChocolate.Skimmed.Serialization;
+
+namespace HotChocolate.Composition.PreMergeValidation.Rules;
+
+public sealed class ExternalArgumentsDefaultMismatchRuleTests
+{
+    [Test]
+    [MethodDataSource(nameof(ValidExamplesData))]
+    public async Task Examples_Valid(string[] sdl)
+    {
+        // arrange
+        var log = new CompositionLog();
+        var context = new CompositionContext([.. sdl.Select(SchemaParser.Parse)], log);
+        var preMergeValidator = new PreMergeValidator([new ExternalArgumentsDefaultMismatchRule()]);
+
+        // act
+        var result = preMergeValidator.Validate(context);
+
+        // assert
+        await Assert.That(result.IsSuccess).IsTrue();
+        await Assert.That(log.IsEmpty).IsTrue();
+    }
+
+    [Test]
+    [MethodDataSource(nameof(InvalidExamplesData))]
+    public async Task Examples_Invalid(string[] sdl)
+    {
+        // arrange
+        var log = new CompositionLog();
+        var context = new CompositionContext([.. sdl.Select(SchemaParser.Parse)], log);
+        var preMergeValidator = new PreMergeValidator([new ExternalArgumentsDefaultMismatchRule()]);
+
+        // act
+        var result = preMergeValidator.Validate(context);
+
+        // assert
+        await Assert.That(result.IsFailure).IsTrue();
+        await Assert.That(log.Count()).IsEqualTo(1);
+        await Assert.That(log.First().Code).IsEqualTo("EXTERNAL_ARGUMENT_DEFAULT_MISMATCH");
+        await Assert.That(log.First().Severity).IsEqualTo(LogSeverity.Error);
+    }
+
+    public static IEnumerable<Func<string[]>> ValidExamplesData()
+    {
+        return
+        [
+            // Fields with the same arguments are mergeable.
```

Comment:
```suggestion
            // Here, the `name` field on Product is defined in one source schema and marked as
            // @external in another. The argument `language` has the same default value in both
            // source schemas, satisfying the rule.
```

When an example comes from the spec, I've been copying the description, as it helps the reader to understand how the rule works.

(backticks used when there is ambiguity like "the name field")